pkg/services/sqlstore/apikey.go

```diff
@@ -38,6 +39,21 @@ func (ss *SQLStore) GetAPIKeys(ctx context.Context, query *models.GetApiKeysQuer
 	})
 }
 
+// GetAPIKeys queries the database based
+// on input on GetApiKeysQuery
+func (ss *SQLStore) GetNonServiceAccountAPIKeys(ctx context.Context) []*models.ApiKey {
+	result := make([]*models.ApiKey, 0)
+	err := ss.WithDbSession(ctx, func(dbSession *DBSession) error {
+		sess := dbSession. //CHECK how many API keys do our clients have?  Can we load them all?
+					Where("(expires IS NULL OR expires >= ?) AND service_account_id < 1 ", timeNow().Unix()).Asc("name")
+		return sess.Find(&result)
+	})
+	if err != nil {
+		ss.log.Warn("API key not loaded", "err", err)
+	}
```

Comment:
we continue here from this if statement, is that supposed to happen here?

just wondering if there is some fault in having a return empty slice here

Reply:
Can you think of a better way?

Reply:
I am not sure here, potentially include a error here. but I know you do not like that right 🤔 

ideally I do not want to return results. but I guess results will just be empty in this case

Reply:
Yes, an empty list will result in no upgrades.

It's not an error condition because having no upgrades will be the normal situation.